unittest/MultiVectorIndex.h

```diff
@@ -0,0 +1,139 @@
+// Copyright (C) 2019-2020 Zilliz. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance
+// with the License. You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distributed under the License
+// is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+// or implied. See the License for the specific language governing permissions and limitations under the License
+
+#include <future>
+
+#include "common/Log.h"
+#include "knowhere/index/vector_index/IndexAnnoy.h"
+#include "knowhere/index/vector_index/IndexBinaryIDMAP.h"
+#include "knowhere/index/vector_index/IndexBinaryIVF.h"
+#include "knowhere/index/vector_index/IndexHNSW.h"
+#include "knowhere/index/vector_index/IndexIDMAP.h"
+#include "knowhere/index/vector_index/IndexIVF.h"
+#include "knowhere/index/vector_index/IndexIVFPQ.h"
+#include "knowhere/index/vector_index/IndexIVFSQ.h"
+#ifdef KNOWHERE_WITH_DISKANN
+#include "knowhere/index/vector_index/IndexDiskANN.h"
+#include "knowhere/index/vector_index/IndexDiskANNConfig.h"
+#endif
+
+namespace knowhere {
+
+class MultiVectorIndex : public VecIndex {
+ public:
+    MultiVectorIndex(std::string type) {
+        if (type == "annoy") {
+            MultiVectorIndexPtr = std::make_shared<knowhere::IndexAnnoy>();
+        } else if (type == "bin_flat") {
+            MultiVectorIndexPtr = std::make_shared<knowhere::BinaryIDMAP>();
+        } else if (type == "bin_ivf_flat") {
+            MultiVectorIndexPtr = std::make_shared<knowhere::BinaryIVF>();
+        } else if (type == "flat") {
+            MultiVectorIndexPtr = std::make_shared<knowhere::IDMAP>();
+        } else if (type == "hnsw") {
+            MultiVectorIndexPtr = std::make_shared<knowhere::IndexHNSW>();
+        } else if (type == "ivf_flat") {
+            MultiVectorIndexPtr = std::make_shared<knowhere::IVF>();
+        } else if (type == "ivf_pq") {
+            MultiVectorIndexPtr = std::make_shared<knowhere::IVFPQ>();
+        } else if (type == "ivf_sq8") {
+            MultiVectorIndexPtr = std::make_shared<knowhere::IVFSQ>();
+        } else {
+            KNOWHERE_THROW_FORMAT("Invalid index type %s", std::string(type).c_str());
+        }
+    }
+
+#ifdef KNOWHERE_WITH_DISKANN
+    MultiVectorIndex(std::string type, MetricType metric_type, std::shared_ptr<FileManager> file_manager) {
```

Comment:
Did you tried `MultiVectorIndex(std::string type, ...)`?

Reply:
I tried, but it will reduce some cite promblem for reason that, the IndexDiskANN.h was separate from the other indexes when build.

Reply:
But I did some changes in the python api , it only has one function to build multivector by using python.